tests/lib/edge_impulse/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+file(REMOVE ${CMAKE_CURRENT_SOURCE_DIR}/src/edge_impulse_dummy.zip)
+add_custom_target(create_zip COMMAND
+    ${CMAKE_COMMAND} -E tar "cf"
+    "../edge_impulse_dummy.zip" --format=zip
```

Comment:
I don't think we should construct zip files in the repo source tree.

As minimum all generated files should be placed in the build folder.

A second note, doing something like this at cmake configure time is bound to break, because the zip file will only be re-created when CMake re-runs, not when you update any of the input files or add new files to the archive.

It might be a minor problem in this code itself, as it will only impact anyone updating the tests at a later stage.
But a bigger risk is users looking into this code being and thinking of it as a reference and thus copying the principle to another location without realizing the side effect it may have.

Reply:
Adding proper dependency would require globbing I think (we compress all of the files that are part of the directory). We also should not explicitly list files in CMakeLists I think.

The zip file is generated every time I start the build (every time I call the ninja). Seems like zipping files is considered a phony target so operation runs during every build. Since it's an unit test, I would say that we could leave it as it is now. Is it also accepted by you?